servers/rendering/renderer_rd/storage_rd/material_storage.cpp

```diff
@@ -987,17 +989,17 @@ void MaterialStorage::MaterialData::free_parameters_uniform_set(RID p_uniform_se
 }
 
 bool MaterialStorage::MaterialData::update_parameters_uniform_set(const HashMap<StringName, Variant> &p_parameters, bool p_uniform_dirty, bool p_textures_dirty, const HashMap<StringName, ShaderLanguage::ShaderNode::Uniform> &p_uniforms, const uint32_t *p_uniform_offsets, const Vector<ShaderCompiler::GeneratedCode::Texture> &p_texture_uniforms, const HashMap<StringName, HashMap<int, RID>> &p_default_texture_params, uint32_t p_ubo_size, RID &uniform_set, RID p_shader, uint32_t p_shader_uniform_set, bool p_use_linear_color, bool p_3d_material) {
-	if ((uint32_t)ubo_data.size() != p_ubo_size) {
+	if ((uint32_t)ubo_data[p_use_linear_color].size() != p_ubo_size) {
```

Comment:
I know the feature works because you reproduced so I'm just trying to learn here.

If `ubo_data[0]` is linear and `ubo_data[1]` is sRGB, why is the boolean `p_use_linear_color` used like this? When it's true, it evaluates to 1 and accesses the sRGB data. Which seems counter-intuitive given the name of the variable.